terraform_1.14.0-alpha20250827 panic: runtime error: invalid memory address or nil pointer dereference

### Terraform Version

```shell
Terraform v1.14.0-alpha20250827
on darwin_arm64
```

### Terraform Configuration Files

https://github.com/dleehr/tfa-demo/tree/main/terraform

### Debug Output

[terraform-crash.txt](https://github.com/user-attachments/files/22012807/terraform-crash.txt)

### Expected Behavior

`terraform plan` should not panic. I have a local build of 3b12eb08667d9ad922fa73eb7765767cefb13c76 that does not panic

### Actual Behavior

```
Terraform will perform the following actions:

  # aap_group.infra will be created
  + resource "aap_group" "infra" {
      + id           = (known after apply)
      + inventory_id = 1
      + name         = "infra"
      + url          = (known after apply)
    }
2025-08-27T15:38:27.543-0400 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2025-08-27T15:38:27.543-0400 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

panic: runtime error: invalid memory address or nil pointer dereference
goroutine 22 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:26 +0x64
github.com/hashicorp/terraform/internal/logging.PanicHandler()
	github.com/hashicorp/terraform/internal/logging/panic.go:84 +0x16c
panic({0x103af8620?, 0x105ff7fb0?})
	runtime/panic.go:783 +0x120
github.com/hashicorp/terraform/internal/command/jsonformat.renderActionInvocation({0x1400089ad80?, 0x140005a4ed0?, 0x21?}, {{{0x14000f3c360, 0x1e}, {0x140001c47e0, 0x11}, {0x140008163f6, 0x5}, 0x14000f60b70, ...}, ...})
	github.com/hashicorp/terraform/internal/command/jsonformat/plan.go:431 +0x13c
github.com/hashicorp/terraform/internal/command/jsonformat.renderHumanDiff({_, _, _}, {{{0x14000f08c80, 0x10}, {0x0, 0x0}, {0x0, 0x0}, {0x102f549e4, ...}, ...}, ...}, ...)
	github.com/hashicorp/terraform/internal/command/jsonformat/plan.go:413 +0x568
github.com/hashicorp/terraform/internal/command/jsonformat.Plan.renderHuman({{0x102f4e71c, 0x3}, 0x14000f38b10, {0x14000273108, 0x6, 0x8}, {0x0, 0x0, 0x0}, {0x14000f152c0, ...}, ...}, ...)
	github.com/hashicorp/terraform/internal/command/jsonformat/plan.go:235 +0xbb4
github.com/hashicorp/terraform/internal/command/jsonformat.Renderer.RenderHumanPlan({0x1400089ad80?, 0x140005a4ed0?, 0x15?}, {{0x102f4e71c, 0x3}, 0x14000f38b10, {0x14000273108, 0x6, 0x8}, {0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform/internal/command/jsonformat/renderer.go:128 +0x148
github.com/hashicorp/terraform/internal/command/views.(*OperationHuman).Plan(0x140001b9a60, 0x14001235dc0, 0x14000f291c0)
	github.com/hashicorp/terraform/internal/command/views/operation.go:131 +0x230
github.com/hashicorp/terraform/internal/backend/local.(*Local).opPlan(0x14000902460, {0x1041da5b0, 0x1400081a500}, {0x1041da5b0, 0x1400081a550}, 0x140007fa780, 0x1400092cb40)
	github.com/hashicorp/terraform/internal/backend/local/backend_plan.go:216 +0xaac
github.com/hashicorp/terraform/internal/backend/local.(*Local).Operation.func1()
	github.com/hashicorp/terraform/internal/backend/local/backend.go:348 +0xbc
created by github.com/hashicorp/terraform/internal/backend/local.(*Local).Operation in goroutine 1
	github.com/hashicorp/terraform/internal/backend/local/backend.go:341 +0x388
```

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Additional Context

_No response_

### References

_No response_

### Generative AI / LLM assisted development?

_No response_

--- a/internal/command/jsonformat/plan_test.go
+++ b/internal/command/jsonformat/plan_test.go
@@ -8544,7 +8544,7 @@ func TestResourceChange_actions(t *testing.T) {
 					},
 				},
 			}
-			jsonschemas := jsonprovider.MarshalForRenderer(fullSchema, true)
+			jsonschemas := jsonprovider.MarshalForRenderer(fullSchema, false)
 			diffs := precomputeDiffs(Plan{
 				ResourceChanges:   []jsonplan.ResourceChange{defaultResourceChange},
 				ActionInvocations: tc.actionInvocations,

--- a/internal/command/jsonprovider/provider.go
+++ b/internal/command/jsonprovider/provider.go
@@ -68,6 +68,7 @@ func marshalProvider(tps providers.ProviderSchema, includeExperimentalSchemas bo
 		EphemeralResourceSchemas: marshalSchemas(tps.EphemeralResourceTypes),
 		Functions:                jsonfunction.MarshalProviderFunctions(tps.Functions),
 		ResourceIdentitySchemas:  marshalIdentitySchemas(tps.ResourceTypes),
+		ActionSchemas:            marshalActionSchemas(tps.Actions),
 	}
 
 	if includeExperimentalSchemas {
@@ -83,8 +84,6 @@ func marshalProvider(tps providers.ProviderSchema, includeExperimentalSchemas bo
 			}
 		}
 		p.ListResourceSchemas = marshalSchemas(listSchemas)
-
-		p.ActionSchemas = marshalActionSchemas(tps.Actions)
 	}
 
 	return p

--- a/internal/command/jsonprovider/provider_test.go
+++ b/internal/command/jsonprovider/provider_test.go
@@ -33,6 +33,7 @@ func TestMarshalProvider(t *testing.T) {
 				DataSourceSchemas:        map[string]*Schema{},
 				EphemeralResourceSchemas: map[string]*Schema{},
 				ResourceIdentitySchemas:  map[string]*IdentitySchema{},
+				ActionSchemas:            map[string]*ActionSchema{},
 			},
 		},
 		{
@@ -212,6 +213,7 @@ func TestMarshalProvider(t *testing.T) {
 					},
 				},
 				ResourceIdentitySchemas: map[string]*IdentitySchema{},
+				ActionSchemas:           map[string]*ActionSchema{},
 			},
 		},
 		{

